packages/ui-shared/lib/components/MultiSelect.vue

```diff
@@ -0,0 +1,98 @@
+<script setup lang="ts">
+import { PropType, computed, ref } from 'vue'
+
+const props = defineProps({
+  showSearch: Boolean,
+
+  options: {
+    type: Object as PropType<Record<string, string>>,
+    required: true
+  },
+
+  modelValue: {
+    type: Array as PropType<string[]>,
+    required: true
+  },
+
+  inputClasses: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  inputWrapperClasses: {
+    type: String,
+    required: false,
+    default: ''
+  },
+
+  contentClasses: {
+    type: String,
+    required: false,
+    default: ''
+  }
+})
+
+const search = ref('')
+
+const emits = defineEmits<{
+  (e: 'update:modelValue', msg: string[]): void
+}>()
+
+const filteredItems = computed(() => {
+  return Object.entries(props.options).filter(([_, value]) =>
+    value.toLowerCase().includes(search.value.toLowerCase())
+  )
+})
+
+function handleToggleItem(key: string) {
+  if (props.modelValue.includes(key)) {
+    emits(
+      'update:modelValue',
+      props.modelValue.filter((message) => message !== key)
+    )
+  } else {
+    emits('update:modelValue', [...props.modelValue!, key])
+  }
+}
+</script>
+
+<template>
+  <div>
+    <BaseDropdown>
+      <template #default="{ shown }">
+        <slot name="default" v-bind="{ shown }" />
+      </template>
+
+      <template #content>
+        <div :class="contentClasses">
+          <div v-if="showSearch" :class="inputWrapperClasses">
+            <input
+              v-model="search"
+              placeholder="Search.."
+              :class="inputClasses"
+              type="text"
+            />
+          </div>
```

Comment:
Lets abstract everything related to the input (showSearch, inputWrapperClasses and search to the parent component) and:
- use a slot here
- pass the search value as a prop to this component